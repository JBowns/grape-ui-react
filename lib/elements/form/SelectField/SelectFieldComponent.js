'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SelectFieldComponent = undefined;

var _templateObject = _taggedTemplateLiteral(['\n    ', '\n    ', '\n    ', '\n    ', '\n  '], ['\n    ', '\n    ', '\n    ', '\n    ', '\n  ']);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _textStyles = require('../../../elements/typography/textStyles');

var _baseControlStyle = require('../ControlGroup/baseControlStyle');

var _componentHelpers = require('../../../utils/componentHelpers');

var _reactSelect = require('react-select');

var _reactSelect2 = _interopRequireDefault(_reactSelect);

var _stateManager = require('react-select/lib/stateManager');

var _stateManager2 = _interopRequireDefault(_stateManager);

var _styledSystem = require('styled-system');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var SelectFieldComponent = exports.SelectFieldComponent = function SelectFieldComponent(props) {
  var chipBg = props.chipBg,
      isDisabled = props.isDisabled,
      isFocused = props.isFocused,
      menuFocusBg = props.menuFocusBg,
      menuFocusColor = props.menuFocusColor,
      menuSelectedBg = props.menuSelectedBg,
      menuSelectedColor = props.menuSelectedColor,
      multiple = props.multiple,
      placeholderColor = props.placeholderColor,
      validationError = props.validationError;

  var textBase = (0, _textStyles.textStylesBase)(props);
  var getControlStyle = function getControlStyle() {
    if (!validationError && !isDisabled) {
      return (0, _baseControlStyle.controlStylesBase)(props);
    } else if (validationError) {
      return (0, _baseControlStyle.controlStylesBase)(Object.assign({}, props, { borderColor: (0, _componentHelpers.resolveColor)('brandDanger'), activeColor: (0, _componentHelpers.resolveColor)('brandDanger') }));
    }
    return (0, _baseControlStyle.controlStylesBase)(Object.assign({}, props, { borderColor: (0, _componentHelpers.resolveColor)('white.light'), activeColor: (0, _componentHelpers.resolveColor)('white.light') }));
  };
  var controlBase = getControlStyle();
  var getFocusStyle = function getFocusStyle() {
    if (isFocused) {
      return (0, _baseControlStyle.focusStylesBase)(props);
    }
    return '';
  };
  var focusBase = getFocusStyle();
  var reactSelectStylesOverrides = '\n    .grape-ui-select__control {\n      display: flex;\n    }\n    .grape-ui-select__indicator {\n      padding: 0;\n    }\n    .grape-ui-select__indicator-separator {\n      display: none;\n    }\n    .grape-ui-select__value-container {\n      padding: 0;\n    }\n    .grape-ui-select__menu {\n      position: absolute;\n      left: 0;\n      margin: 0;\n      padding: 0.5rem 0;\n      ' + (0, _componentHelpers.resolveElevation)('03') + '\n    }\n    .grape-ui-select__option {\n      padding: 0.5rem 1rem;\n    }\n    .grape-ui-select__option--is-focused {\n      background: ' + (0, _componentHelpers.resolveColor)(menuFocusBg) + ';\n      color: ' + (0, _componentHelpers.resolveColor)(menuFocusColor) + ';\n    }\n    .grape-ui-select__option--is-selected {\n      background: ' + (0, _componentHelpers.resolveColor)(menuSelectedBg) + ';\n      color: ' + (0, _componentHelpers.resolveColor)(menuSelectedColor) + ';\n    }\n    .grape-ui-select__placeholder {\n      color: ' + (0, _componentHelpers.resolveColor)(placeholderColor) + ';\n    }\n    .grape-ui-select__multi-value {\n      display: flex;\n      margin-right: 0.25rem;\n      padding: 0.25rem;\n      border-radius: 4px;\n      background-color: ' + (0, _componentHelpers.resolveColor)(chipBg) + ';\n      font-size: 80%;\n    }\n    .grape-ui-select__multi-value__label {\n      padding: 0 0.25rem;\n    }\n    .grape-ui-select__multi-value__remove {\n      cursor: pointer;\n      &:hover path {\n        fill: ' + (0, _componentHelpers.resolveColor)('brandLinkHover') + '\n      }\n    }\n  ';

  var styleOverrides = {
    control: function control(styles) {
      return null;
    },
    option: function option(styles) {
      return null;
    },
    input: function input(styles) {
      return null;
    },
    multiValue: function multiValue(styles) {
      return null;
    },
    multiValueLabel: function multiValueLabel(styles) {
      return null;
    },
    multiValueRemove: function multiValueRemove(styles) {
      return null;
    },
    placeholder: function placeholder(styles) {
      return null;
    },
    singleValue: function singleValue(styles) {
      return null;
    }
  };

  var ProtoBase = (0, _styledComponents2.default)(_reactSelect2.default).attrs({
    isDisabled: isDisabled,
    isMulti: multiple,
    styles: styleOverrides,
    className: 'grape-ui-select-container',
    classNamePrefix: 'grape-ui-select'
  })(_templateObject, textBase, controlBase, focusBase, reactSelectStylesOverrides);

  return _react2.default.createElement(ProtoBase, (0, _componentHelpers.passThrough)(SelectFieldComponent, props));
};

SelectFieldComponent.propTypes = {
  activeColor: _propTypes2.default.string,
  borderColor: _propTypes2.default.string,
  borderRadius: _propTypes2.default.string,
  chipBg: _propTypes2.default.string,
  color: _propTypes2.default.string,
  fontFamily: _propTypes2.default.string,
  fontWeight: _propTypes2.default.string,
  isDisabled: _propTypes2.default.bool,
  isFocused: _propTypes2.default.bool,
  kerning: _propTypes2.default.string,
  lg: _propTypes2.default.bool,
  menuFocusBg: _propTypes2.default.string,
  menuFocusColor: _propTypes2.default.string,
  menuSelectedBg: _propTypes2.default.string,
  menuSelectedColor: _propTypes2.default.string,
  multiple: _propTypes2.default.bool,
  padding: _propTypes2.default.string,
  placeholderColor: _propTypes2.default.string,
  sm: _propTypes2.default.bool,
  textAlign: _propTypes2.default.string,
  textDecoration: _propTypes2.default.string,
  validationError: _propTypes2.default.string
};

SelectFieldComponent.defaultProps = {
  activeColor: _baseControlStyle.defaultControlStylesBase.activeColor,
  borderColor: _baseControlStyle.defaultControlStylesBase.borderColor,
  borderRadius: _baseControlStyle.defaultControlStylesBase.borderRadius,
  chipBg: 'white.dark',
  color: _textStyles.defaultTextStylesBase.color,
  fontFamily: _textStyles.defaultTextStylesBase.fontFamily,
  fontWeight: _textStyles.defaultTextStylesBase.fontWeight,
  kerning: _textStyles.defaultTextStylesBase.kerning,
  isDisabled: false,
  isFocused: _baseControlStyle.defaultControlStylesBase.isFocused,
  lg: _textStyles.defaultTextStylesBase.lg,
  menuFocusBg: 'brandLinkHover',
  menuFocusColor: 'white',
  menuSelectedBg: 'brandLink',
  menuSelectedColor: 'white',
  multiple: false,
  padding: _baseControlStyle.defaultControlStylesBase.padding,
  placeholderColor: _baseControlStyle.defaultControlStylesBase.placeholderColor,
  sm: _textStyles.defaultTextStylesBase.sm,
  textAlign: _textStyles.defaultTextStylesBase.textAlign,
  textDecoration: _textStyles.defaultTextStylesBase.textDecoration,
  validationError: ''
};