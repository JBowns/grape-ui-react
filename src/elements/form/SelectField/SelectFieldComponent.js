import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { defaultTextStylesBase, textStylesBase } from 'src/elements/typography/textStyles';
import { defaultControlStylesBase, controlStylesBase } from '../ControlGroup/baseControlStyle';
import { passThrough, resolveColor } from 'src/utils/componentHelpers';
import Select from 'react-select';

export const SelectFieldComponent = props => {
  const { multiple, plainText, validationError } = props;
  const textBase = textStylesBase(props);
  const controlBase = !validationError ? controlStylesBase(props) : controlStylesBase({ ...props, borderColor: resolveColor('brandDanger'), activeColor: resolveColor('brandDanger') });
  const theme = () => ({
    borderRadius: 0,
    colors: {
      text: props.color,
      primary25: 'hotpink',
      primary: 'black',
    },
  });


  const ProtoBase = styled(Select).attrs({
    theme: {theme},
    readOnly: plainText,
    tabIndex: !plainText ? '0' : '-1',
    isMulti: multiple,
  })`
    ${textBase}
    ${controlBase}
  `;

  return <ProtoBase {...passThrough(SelectFieldComponent, props)} />;
};

SelectFieldComponent.propTypes = {
  activeColor: PropTypes.string,
  borderColor: PropTypes.string,
  borderRadius: PropTypes.string,
  color: PropTypes.string,
  fontFamily: PropTypes.string,
  fontWeight: PropTypes.string,
  kerning: PropTypes.string,
  lg: PropTypes.bool,
  multiple: PropTypes.bool,
  padding: PropTypes.string,
  plainText: PropTypes.bool,
  sm: PropTypes.bool,
  textAlign: PropTypes.string,
  textDecoration: PropTypes.string,
  validationError: PropTypes.string,
};

SelectFieldComponent.defaultProps = {
  activeColor: defaultControlStylesBase.activeColor,
  borderColor: defaultControlStylesBase.borderColor,
  borderRadius: defaultControlStylesBase.borderRadius,
  color: defaultTextStylesBase.color,
  fontFamily: defaultTextStylesBase.fontFamily,
  fontWeight: defaultTextStylesBase.fontWeight,
  kerning: defaultTextStylesBase.kerning,
  lg: defaultTextStylesBase.lg,
  multiple: false,
  padding: defaultControlStylesBase.padding,
  plainText: defaultControlStylesBase.plainText,
  sm: defaultTextStylesBase.sm,
  textAlign: defaultTextStylesBase.textAlign,
  textDecoration: defaultTextStylesBase.textDecoration,
  validationError: '',
};
