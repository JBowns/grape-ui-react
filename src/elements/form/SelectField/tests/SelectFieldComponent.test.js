import React from 'react';
import { SelectFieldComponent } from '../SelectFieldComponent';
import 'jest-styled-components';
// import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { configure, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });
const colorOptions = [
  { value: 'red', label: 'Red', color: '#FF5630' },
  { value: 'yellow', label: 'Yellow', color: '#FFC400' },
  { value: 'green', label: 'Green', color: '#36B37E' },
];

describe('SelectFieldComponent base', () => {
  it('should return a SelectFieldComponent object', () => {
    const component = mount(<SelectFieldComponent defaultValue={colorOptions[1]} id="exampleColor" options={colorOptions} />);
    expect(toJson(component.find('SelectFieldComponent'))).toMatchSnapshot();
  });
});

describe('SelectFieldComponent with some style relate', () => {
  it('should return a SelectFieldComponent lg object', () => {
    const component = mount(<SelectFieldComponent defaultValue={colorOptions[1]} id="exampleColor" lg options={colorOptions} />);
    expect(toJson(component.find('SelectFieldComponent'))).toMatchSnapshot();
  });
  it('should return a SelectFieldComponent sm object', () => {
    const component = mount(<SelectFieldComponent defaultValue={colorOptions[1]} id="exampleColor" options={colorOptions} sm />);
    expect(toJson(component.find('SelectFieldComponent'))).toMatchSnapshot();
  });
});

