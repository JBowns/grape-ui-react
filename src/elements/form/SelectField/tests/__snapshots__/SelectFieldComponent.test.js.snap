// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectFieldComponent base should return a SelectFieldComponent object 1`] = `
.c0 {
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  border: 1px solid hsl(0,0%,25%);
  border-radius: 4px;
  border-radius: 4px;
  outline: 0;
  width: 100%;
  padding: 1rem;
}

.c0 .grape-ui-select__control {
  margin: -1rem;
  padding: 1rem;
}

.c0[disabled] {
  opacity: 0.3;
}

.c0[disabled] ~ * {
  color: rgba(0,0,0,0.3);
}

.c0:focus,
.c0:focus-within {
  border-color: hsl(323.3,84.6%,28%);
  box-shadow: 0 0 0 1px hsl(323.3,84.6%,28%);
}

.c0:focus + label,
.c0:focus-within + label {
  color: hsl(323.3,84.6%,28%);
}

.c0::-webkit-input-placeholder {
  color: hsl(0,0%,75%);
}

.c0::-moz-placeholder {
  color: hsl(0,0%,75%);
}

.c0:-ms-input-placeholder {
  color: hsl(0,0%,75%);
}

.c0::placeholder {
  color: hsl(0,0%,75%);
}

<div
  className="css-1pcexqc-container c0"
  id="exampleColor"
  onKeyDown={[Function]}
>
  <div
    className="css-mazh0z grape-ui-select__control"
    onMouseDown={[Function]}
    onTouchEnd={[Function]}
  >
    <div
      className="css-14cvlpc grape-ui-select__value-container grape-ui-select__value-container--has-value"
    >
      <div
        className="css-8uhtka grape-ui-select__single-value"
      >
        Yellow
      </div>
      <div
        className="css-0"
      >
        <div
          className="grape-ui-select__input"
          style={
            Object {
              "display": "inline-block",
            }
          }
        >
          <input
            aria-autocomplete="list"
            autoCapitalize="none"
            autoComplete="off"
            autoCorrect="off"
            disabled={false}
            id="react-select-2-input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            spellCheck="false"
            style={
              Object {
                "background": 0,
                "border": 0,
                "boxSizing": "content-box",
                "color": "inherit",
                "fontSize": "inherit",
                "label": "input",
                "opacity": 1,
                "outline": 0,
                "padding": 0,
                "width": "1px",
              }
            }
            tabIndex="0"
            type="text"
            value=""
          />
          <div
            style={
              Object {
                "height": 0,
                "left": 0,
                "overflow": "scroll",
                "position": "absolute",
                "top": 0,
                "visibility": "hidden",
                "whiteSpace": "pre",
              }
            }
          >
            
          </div>
        </div>
      </div>
    </div>
    <div
      className="css-1rtnzd2 grape-ui-select__indicators"
    >
      <span
        className="css-1hyfx7x grape-ui-select__indicator-separator"
      />
      <div
        aria-hidden="true"
        className="css-1pseabs-indicatorContainer grape-ui-select__indicator grape-ui-select__dropdown-indicator"
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
      >
        <svg
          aria-hidden="true"
          className="css-19bqh2r"
          focusable="false"
          height={20}
          viewBox="0 0 20 20"
          width={20}
        >
          <path
            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`SelectFieldComponent with some style relate should return a SelectFieldComponent lg object 1`] = `
.c0 {
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 1.2rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  border: 1px solid hsl(0,0%,25%);
  border-radius: 8px;
  border-radius: 8px;
  outline: 0;
  width: 100%;
  padding: 1rem;
}

.c0 .grape-ui-select__control {
  margin: -1rem;
  padding: 1rem;
}

.c0[disabled] {
  opacity: 0.3;
}

.c0[disabled] ~ * {
  color: rgba(0,0,0,0.3);
}

.c0:focus,
.c0:focus-within {
  border-color: hsl(323.3,84.6%,28%);
  box-shadow: 0 0 0 1px hsl(323.3,84.6%,28%);
}

.c0:focus + label,
.c0:focus-within + label {
  color: hsl(323.3,84.6%,28%);
}

.c0::-webkit-input-placeholder {
  color: hsl(0,0%,75%);
}

.c0::-moz-placeholder {
  color: hsl(0,0%,75%);
}

.c0:-ms-input-placeholder {
  color: hsl(0,0%,75%);
}

.c0::placeholder {
  color: hsl(0,0%,75%);
}

<div
  className="css-1pcexqc-container c0"
  id="exampleColor"
  onKeyDown={[Function]}
>
  <div
    className="css-mazh0z grape-ui-select__control"
    onMouseDown={[Function]}
    onTouchEnd={[Function]}
  >
    <div
      className="css-14cvlpc grape-ui-select__value-container grape-ui-select__value-container--has-value"
    >
      <div
        className="css-8uhtka grape-ui-select__single-value"
      >
        Yellow
      </div>
      <div
        className="css-0"
      >
        <div
          className="grape-ui-select__input"
          style={
            Object {
              "display": "inline-block",
            }
          }
        >
          <input
            aria-autocomplete="list"
            autoCapitalize="none"
            autoComplete="off"
            autoCorrect="off"
            disabled={false}
            id="react-select-3-input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            spellCheck="false"
            style={
              Object {
                "background": 0,
                "border": 0,
                "boxSizing": "content-box",
                "color": "inherit",
                "fontSize": "inherit",
                "label": "input",
                "opacity": 1,
                "outline": 0,
                "padding": 0,
                "width": "1px",
              }
            }
            tabIndex="0"
            type="text"
            value=""
          />
          <div
            style={
              Object {
                "height": 0,
                "left": 0,
                "overflow": "scroll",
                "position": "absolute",
                "top": 0,
                "visibility": "hidden",
                "whiteSpace": "pre",
              }
            }
          >
            
          </div>
        </div>
      </div>
    </div>
    <div
      className="css-1rtnzd2 grape-ui-select__indicators"
    >
      <span
        className="css-1hyfx7x grape-ui-select__indicator-separator"
      />
      <div
        aria-hidden="true"
        className="css-1pseabs-indicatorContainer grape-ui-select__indicator grape-ui-select__dropdown-indicator"
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
      >
        <svg
          aria-hidden="true"
          className="css-19bqh2r"
          focusable="false"
          height={20}
          viewBox="0 0 20 20"
          width={20}
        >
          <path
            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`SelectFieldComponent with some style relate should return a SelectFieldComponent sm object 1`] = `
.c0 {
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 0.8rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  border: 1px solid hsl(0,0%,25%);
  border-radius: 2px;
  border-radius: 2px;
  outline: 0;
  width: 100%;
  padding: 1rem;
}

.c0 .grape-ui-select__control {
  margin: -1rem;
  padding: 1rem;
}

.c0[disabled] {
  opacity: 0.3;
}

.c0[disabled] ~ * {
  color: rgba(0,0,0,0.3);
}

.c0:focus,
.c0:focus-within {
  border-color: hsl(323.3,84.6%,28%);
  box-shadow: 0 0 0 1px hsl(323.3,84.6%,28%);
}

.c0:focus + label,
.c0:focus-within + label {
  color: hsl(323.3,84.6%,28%);
}

.c0::-webkit-input-placeholder {
  color: hsl(0,0%,75%);
}

.c0::-moz-placeholder {
  color: hsl(0,0%,75%);
}

.c0:-ms-input-placeholder {
  color: hsl(0,0%,75%);
}

.c0::placeholder {
  color: hsl(0,0%,75%);
}

<div
  className="css-1pcexqc-container c0"
  id="exampleColor"
  onKeyDown={[Function]}
>
  <div
    className="css-mazh0z grape-ui-select__control"
    onMouseDown={[Function]}
    onTouchEnd={[Function]}
  >
    <div
      className="css-14cvlpc grape-ui-select__value-container grape-ui-select__value-container--has-value"
    >
      <div
        className="css-8uhtka grape-ui-select__single-value"
      >
        Yellow
      </div>
      <div
        className="css-0"
      >
        <div
          className="grape-ui-select__input"
          style={
            Object {
              "display": "inline-block",
            }
          }
        >
          <input
            aria-autocomplete="list"
            autoCapitalize="none"
            autoComplete="off"
            autoCorrect="off"
            disabled={false}
            id="react-select-4-input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            spellCheck="false"
            style={
              Object {
                "background": 0,
                "border": 0,
                "boxSizing": "content-box",
                "color": "inherit",
                "fontSize": "inherit",
                "label": "input",
                "opacity": 1,
                "outline": 0,
                "padding": 0,
                "width": "1px",
              }
            }
            tabIndex="0"
            type="text"
            value=""
          />
          <div
            style={
              Object {
                "height": 0,
                "left": 0,
                "overflow": "scroll",
                "position": "absolute",
                "top": 0,
                "visibility": "hidden",
                "whiteSpace": "pre",
              }
            }
          >
            
          </div>
        </div>
      </div>
    </div>
    <div
      className="css-1rtnzd2 grape-ui-select__indicators"
    >
      <span
        className="css-1hyfx7x grape-ui-select__indicator-separator"
      />
      <div
        aria-hidden="true"
        className="css-1pseabs-indicatorContainer grape-ui-select__indicator grape-ui-select__dropdown-indicator"
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
      >
        <svg
          aria-hidden="true"
          className="css-19bqh2r"
          focusable="false"
          height={20}
          viewBox="0 0 20 20"
          width={20}
        >
          <path
            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
